components/helm-broker/internal/broker/provision.go

```diff
@@ -36,9 +39,9 @@ type provisionService struct {
 	testHookAsyncCalled func(internal.OperationID)
 }
 
-func (svc *provisionService) Provision(ctx context.Context, osbCtx OsbContext, req *osb.ProvisionRequest) (*osb.ProvisionResponse, error) {
+func (svc *provisionService) Provision(ctx context.Context, osbCtx OsbContext, req *osb.ProvisionRequest) (*osb.ProvisionResponse, osb.HTTPStatusCodeError) {
```

Comment:
let's return (*osb.ProvisionResponse, *osb.HTTPStatusCodeError) and the caller can check if the error is nil